plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -575,6 +577,12 @@ public static void createDirectory(HdfsContext context, HdfsEnvironment hdfsEnvi
                 throw new IOException("mkdirs returned false");
             }
         }
+        catch (FileAlreadyExistsException ex) {
+            throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed to create directory, destination exists and is not a directory: " + path, ex);
```

Comment:
Can we rename the word as `destination is a file`

Reply:
refactored